apps/meteor/client/views/invite/InvitePage.tsx

```diff
@@ -60,29 +80,17 @@ const InvitePage = (): ReactElement => {
 					return;
 				}
 
-				try {
-					const result = await sdk.rest.post('/v1/useInviteToken', { token });
-
-					if (!result.room.name) {
-						dispatchToastMessage({ type: 'error', message: t('Failed_to_activate_invite_token') });
-						homeRoute.push();
-						return;
-					}
-
-					if (result.room.t === 'p') {
-						groupRoute.push({ name: result.room.name });
-						return;
-					}
-
-					channelRoute.push({ name: result.room.name });
-				} catch (error) {
-					dispatchToastMessage({ type: 'error', message: t('Failed_to_activate_invite_token') });
-					homeRoute.push();
-				}
+				return inviteTokenMutation.mutate(token);
 			},
 		},
 	);
 
+	useEffect(() => {
+		if (userId && token) {
+			inviteTokenMutation.mutate(token);
+		}
+	}, [inviteTokenMutation, token, userId]);
+
```

Comment:
I think that because of this is running the mutation several times...
![image](https://github.com/RocketChat/Rocket.Chat/assets/20212776/62a0d555-b012-44e9-9fe3-3f597c35e31e)

Reply:
For the record: The mutation object returned by the useMutation hook is not stable and will change many times in each re-render of the component, causing the problem described above, the .mutate function on the other hand is stable and simply destructuring it from the returned object and using it in the useEffect solved the issue.